TransIT.BLL/Services/UserService.cs

```
@@ -60,69 +70,102 @@ public class UserService : IUserService
-            value.Password = _hasher.HashPassword(value.Password);
-            
-            await _userRepository.AddAsync(value);
-            await _uow.SaveAsync();
-            return true;
+            user.Id = 0;
+            user.Password = _hasher.HashPassword(user.Password);
+
+            try
+            {
+                var res = await _userRepository.AddAsync(user);
+                await _unitOfWork.SaveAsync();
+                return res.Entity;
+            }
+            catch (DbUpdateException e)
+            {
+                _logger.LogError(e, nameof(Create), e.Entries);
+                return null;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, nameof(Create));
+                throw e;
+            }
         }
 
         /// <summary>
         /// Updates user if it is found
         /// </summary>
-        /// <param name="value">User to update</param>
+        /// <param name="user">User to update</param>
         /// <returns>Is successful</returns>
-        public async Task<bool> Update(User value)
+        public async Task<User> Update(User user)
         {
-            if (await _userRepository.GetByIdAsync(value.Id) == null)
-                return false;
+            if (await _userRepository.GetByIdAsync(user.Id) == null)
+                return null;
 
-            _userRepository.Update(value);
-            await _uow.SaveAsync();
-            return true;
+            try
+            {
+                var res = _userRepository.Update(user);
+                await _unitOfWork.SaveAsync();
+                return res.Entity;
+            }
+            catch (DbUpdateException e)
+            {
+                _logger.LogError(e, nameof(Update), e.Entries);
+                return null;
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, nameof(Update));
+                throw e;
+            }
```

Same here
DbUpdateException should catch all exceptions

---

It will catch exceptions only from EF. Point was to stop request if something wrong with other part of app. For example: some cast exception in repo.